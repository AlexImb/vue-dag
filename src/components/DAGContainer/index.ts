import DAGContainer from './DAGContainer.vue';

export { DAGContainer };
