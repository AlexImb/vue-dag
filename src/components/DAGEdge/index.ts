import DAGEdge from './DAGEdge.vue';

export { DAGEdge };
