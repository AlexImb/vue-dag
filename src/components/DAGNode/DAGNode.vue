<template>
  <div class="vd__node">
    <div class="vd__node__link vd__node__link--top"></div>
    <div class="vd__node__link vd__node__link--right"></div>
    <div class="vd__node__link vd__node__link--bottom"></div>
    <div class="vd__node__link vd__node__link--left"></div>
    <slot>
      <div class="vd__node__default-content">
        Empty node
      </div>
    </slot>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class DAGNode extends Vue {}
</script>
