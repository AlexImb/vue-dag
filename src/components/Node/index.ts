import Node from './Node.vue';

export { Node };
