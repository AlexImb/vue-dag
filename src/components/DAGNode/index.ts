import DAGNode from './DAGNode.vue';

export { DAGNode };
