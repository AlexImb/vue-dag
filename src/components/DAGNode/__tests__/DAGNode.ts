import { shallowMount, Wrapper } from '@vue/test-utils';
import { DAGNode } from '@/components/DAGNode';

let wrapper: Wrapper<DAGNode>;

describe('DAGNode', () => {
  beforeEach(() => {
    wrapper = shallowMount(DAGNode);
  });

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
