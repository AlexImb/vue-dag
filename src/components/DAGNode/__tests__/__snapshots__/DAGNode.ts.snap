// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DAGNode renders HelloWorld component inside the default slot 1`] = `
<div class="vd__node" style="transform: scale(1);">
  <div class="vd__node__link vd__node__link--top"></div>
  <div class="vd__node__link vd__node__link--right"></div>
  <div class="vd__node__link vd__node__link--left"></div>
  <div class="vd__node__link vd__node__link--bottom"></div>
  <div>
    <h1>Hello </h1>
  </div>
</div>
`;

exports[`DAGNode renders correctly 1`] = `
<div class="vd__node" style="transform: scale(1);">
  <div class="vd__node__link vd__node__link--top"></div>
  <div class="vd__node__link vd__node__link--right"></div>
  <div class="vd__node__link vd__node__link--left"></div>
  <div class="vd__node__link vd__node__link--bottom"></div>
  <div class="vd__node__default-content">
    Empty node
  </div>
</div>
`;
